common/src/main/scala/azure/APNSRawPush.scala

```diff
@@ -2,11 +2,16 @@ package azure
 
 import azure.apns.Body
 import play.api.http.Writeable
-import play.api.libs.json.{Json, JsValue}
+import play.api.libs.json.{JsValue, Json}
+import play.api.libs.ws.{WSRequest, WSResponse}
 import utils.WriteableImplicits._
 
-case class APNSRawPush(body: Body, tags: Option[Tags]) extends RawPush[Body] {
+import scala.concurrent.Future
+
+case class APNSRawPush(body: Body, tags: Option[Tags]) extends RawPush {
   override def format: String = "apple"
 
-  override def writeable: Writeable[Body] = implicitly[Writeable[JsValue]].map(Json.toJson[Body]).withContentType("application/json;charset=utf-8")
+  private val writeable: Writeable[Body] = implicitly[Writeable[JsValue]].map(Json.toJson[Body]).withContentType("application/json;charset=utf-8")
+
+  override def post(request: WSRequest): Future[WSResponse] = request.post(body)(writeable)
```

Comment:
I ran into issues with RawPush[T] needing to be covariant in T because of `def body: T` and contravariant in T because of ` def writeable: Writeable[T]`.  Moving the post into RawPush removed the need for writeable to be directly exposed in the interface which avoided the problem